### PR TITLE
fix(series): fix 'hasMore' param

### PR DESCRIPTION
**Pull request by AntonLantukh**
## Description

- fix 'hasMore' param

### Steps completed:

According to our definition of done, I have completed the following steps:

- [ ] Acceptance criteria met
- [ ] Unit tests added
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

### CODE REVIEW

File: src/components/CardGrid/CardGrid.tsx [hunk 1]
```diff
@@ -90,9 +91,14 @@ function CardGrid({
   };
 
   return (
-    <InfiniteScroll pageStart={0} loadMore={loadMore || defaultLoadMore} hasMore={hasLoadMore || defaultHasMore} loader={<InfiniteScrollLoader key="loader" />}>
+    <InfiniteScroll
+      pageStart={0}
+      loadMore={isDefined(loadMore) ? loadMore : defaultLoadMore}
```
Comment (budzworthy) [inline src/components/CardGrid/CardGrid.tsx]: Is there any difference to this?
```suggestion
      loadMore={loadMore ?? defaultLoadMore}
```

File: src/components/CardGrid/CardGrid.tsx [hunk 2]
```diff
@@ -90,9 +91,14 @@ function CardGrid({
...
+      hasMore={isDefined(hasMore) ? hasMore : defaultHasMore}
+      loader={<InfiniteScrollLoader key="loader" />}
+    >
       <div className={classNames(styles.container, styles[`cols-${visibleTiles}`])} role="grid">
-        {playlist.playlist.slice(0, rowCount * visibleTiles).map(renderTile)}
+        {(isDefined(loadMore) ? playlist.playlist : playlist.playlist.slice(0, rowCount * visibleTiles)).map(renderTile)}
```
Comment (budzworthy) [inline src/components/CardGrid/CardGrid.tsx]: The logic on this looks backwards to me. Can you explain / maybe comment?

Comment (AntonLantukh) [reply src/components/CardGrid/CardGrid.tsx]: When `loadMore` function is defined, then we use the accumulated data we received from the back-end and don't need to hide some of the cards manually. We hide them manually for playlists where we receive all the data at once (no pagination), but don't want to render cards which are not visible.

Comment (budzworthy) [reply src/components/CardGrid/CardGrid.tsx]: Oh OK I understand. Maybe worth a quick comment in the code. This is the 'fix' for the bug, yes?

Comment (AntonLantukh) [reply src/components/CardGrid/CardGrid.tsx]: Will do. Not only this bug, we also used `defaultHasMore` when we getting `false` from the series endpoint.

File: src/components/CardGrid/CardGrid.tsx (hunk 1 above)
Comment (AntonLantukh) [reply src/components/CardGrid/CardGrid.tsx]: It checks both `null` and `undefined`. My idea was to filter `undefined` props, but I don't really have strong preference here.

Comment (budzworthy) [reply src/components/CardGrid/CardGrid.tsx]: Ooh, I see. OK. I guess it's potentially nice to be able to set it to null, though I'm not sure what the use case would be here specifically.

Comment (AntonLantukh) [reply src/components/CardGrid/CardGrid.tsx]: Yeah, that is why I removed it here and used nullish coalescing operator instead.

**Review by budzworthy — APPROVED** [review]

**Pull request merged**